architecture/core.rst

```diff
@@ -533,6 +557,51 @@ interrupt occurs:
             }
     }
 
+**Using device tree**
+
+Device drivers can rely on Device Tree to get references to the
+interrupt they consume. Refer to the Device Tree documentation
+for how to properly define interrupt controller chip devices and reference
+interrupts in consumer device nodes.
+
+The example in the previous section implementing an interrupt consumer driver
+could be implemented as below, when the platform uses Device Tree means
+to define some platform resources.
+
+.. code-block:: c
+
+    static TEE_Result foo_probe(const void *fdt, int node, const void *compat)
+    {
+            TEE_Result res = TEE_ERROR_GENERIC;
+            struct itr_handler *itr_handler = NULL;
+            struct itr_chip *itr_chip = NULL;
+            size_t itr_num = 0;
+
+            res = dt_get_interrupt(fdt, node, &itr_chip, &itr_num);
+            if (res)
+                    return res;
+
+            res = interrupt_create_handler(&itr_chip, &itr_num,
+                                           foo_it_handler_fn,
+                                           &foo_int1_data,
+                                           &itr_handler);
+            if (res)
+                    return res;
+
+            interrupt_enable(itr_chip, it_num);
+
+            (...)
+
+            return TEE_SUCCESS;
+
+    err_disable_release_irq:
+            interrupt_disable(itr_chip, it_num);
+            interrupt_remove_free_handler(itr_handler);
+
+            return res;
+
+    }
+
```

Comment:
Would it make sense to add the `DEFINE_DT_DRIVER()` part to show how the function is supposed to be called too?

Reply:
OK, i'll add that.